core/src/main/java/org/elasticsearch/common/joda/DateMathParser.java

```diff
@@ -63,13 +63,10 @@ public long parse(String text, Callable<Long> now, boolean roundUp, DateTimeZone
         } else {
             int index = text.indexOf("||");
             if (index == -1) {
-                return parseDateTime(text, timeZone);
+                return parseDateTime(text, timeZone, roundUp);
             }
-            time = parseDateTime(text.substring(0, index), timeZone);
+            time = parseDateTime(text.substring(0, index), timeZone, false);
```

Comment:
I found it weird that we ignore the incoming round up parameter here. Are you sure it's correct?